types/externs.d.ts

```diff
@@ -125,6 +125,9 @@ declare global {
 
     export type OutputMode = 'json' | 'html' | 'csv';
 
+    type DeviceMetricsOverrideParams = Pick<LH.Crdp.Emulation.SetDeviceMetricsOverrideRequest, 'width'|'height'|'deviceScaleFactor'|'mobile'>;
+    export type ScreenEmulationSettings = Partial<DeviceMetricsOverrideParams & {disabled: boolean}>;
```

Comment:
can I pitch you on being explicit (instead of a `Pick<>` and an intersection, which is kind of confusing :) It might help make the options/hierarchy a bit more clear:

```suggestion
    export type ScreenEmulationSettings = {
      /** Overriding width value in pixels (minimum 0, maximum 10000000). 0 disables the override. */
      width: number,
      /** Overriding height value in pixels (minimum 0, maximum 10000000). 0 disables the override. */
      height: number,
      /** Overriding device scale factor value. 0 disables the override. */
      deviceScaleFactor: number,
      /** Whether to emulate mobile device. This includes viewport meta tag, overlay scrollbars, text autosizing and more. */
      mobile: boolean,
      /** Whether screen emulation is disabled. If true, the other emulation settings are ignored. */
      disabled: boolean,
    };
```

The `Emulation.setDeviceMetricsOverride` call in `emulation.js` will make sure the types match `SetDeviceMetricsOverrideRequest`. Downside is having to copy over the doc strings, but they haven't changed since the protocol types were first generated, so seems not bad.

Reply:
love it.